File: app/models/cell.rb
class Cell

  attr_accessor :alive, :x, :y

  def initialize(x=0, y=0)
    @alive = false
    @x = x
    @y = y
  end

  def die!
    alive = false
  end

  def revive!
    alive = true
  end

  def alive?; alive; end

  def dead?; !alive; end
end

File: app/assets/javascripts/app/Cell.js
class Cell {
  constructor(alive, x, y) {
    this.alive = alive;
    this.x = x;
    this.y = y;
  }

  die () { this.alive = false; }

  revive () { this.alive = true; }

  isAlive () { return this.alive; }

  dead () { return !this.alive; }
}

export default Cell;